x11-packages/rnote/build.sh

```diff
@@ -0,0 +1,67 @@
+TERMUX_PKG_HOMEPAGE="https://github.com/flxzt/rnote"
+TERMUX_PKG_DESCRIPTION="An infinite canvas vector-based drawing application for handwritten notes"
+TERMUX_PKG_LICENSE="GPL-3.0"
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_VERSION="0.11.0"
+TERMUX_PKG_SRCURL="https://github.com/flxzt/rnote/archive/v${TERMUX_PKG_VERSION}/v${TERMUX_PKG_VERSION}.tar.gz"
+TERMUX_PKG_SHA256=b133d4331963d3c09d3a7477f60fc4c5072471dcbf459379a593ca1724164af4
+TERMUX_PKG_AUTO_UPDATE=true
+TERMUX_PKG_DEPENDS="gdk-pixbuf, gettext, glib, graphene, gtk4, hicolor-icon-theme, libadwaita, libcairo, pipewire, pango, poppler"
+TERMUX_PKG_BUILD_DEPENDS="libiconv"
+
+__fetch_gettext_rs() {
+    # This version is relative to gettext-sys provided by the gettext-rs crate
+	local _version="0.21.4"
+	local _url="https://github.com/gettext-rs/gettext-rs/archive/refs/tags/gettext-sys-$_version.tar.gz"
+	local _path="$TERMUX_PKG_CACHEDIR/gettext-v$_version.tar.gz"
+	termux_download "$_url" "$_path" SKIP_CHECKSUM
+	tar xf "$_path" -C "$TERMUX_PKG_SRCDIR"
+	mv "gettext-rs-gettext-sys-$_version" "gettext-source"
+
+
+	# Fetch latest and greatest gettext from upstream and place it inside gettext-source/gettext-sys
+	local _upstream_source_version="0.22.5"
+	wget "https://mirrors.kernel.org/gnu/gettext/gettext-${_upstream_source_version}.tar.xz" -P "$TERMUX_PKG_SRCDIR/gettext-source/gettext-sys"
+}
+
+__patch_gettext_rs() {
+    # Patch gettext-rs crate to use a newer gettext version because the old one doesn't compile properly with clang
+    patch -p1 -d "$TERMUX_PKG_SRCDIR/gettext-source" < "$TERMUX_PKG_BUILDER_DIR"/gettext-rs-crate-patch.diff
+}
+
+termux_step_pre_configure() {
+	termux_setup_cmake
+
+	local _WRAPPER_BIN="${TERMUX_PKG_BUILDDIR}/_wrapper/bin"
+	mkdir -p "${_WRAPPER_BIN}"
+	if [[ "${TERMUX_ON_DEVICE_BUILD}" == "false" ]]; then
+		sed "s|^export PKG_CONFIG_LIBDIR=|export PKG_CONFIG_LIBDIR=${TERMUX_PREFIX}/opt/glib/cross/lib/x86_64-linux-gnu/pkgconfig:|" \
+			"${TERMUX_STANDALONE_TOOLCHAIN}/bin/pkg-config" \
+			> "${_WRAPPER_BIN}/pkg-config"
+		chmod +x "${_WRAPPER_BIN}/pkg-config"
+		export PKG_CONFIG="${_WRAPPER_BIN}/pkg-config"
+	fi
+	export PATH="${_WRAPPER_BIN}:${PATH}"
+
+	# Fetch and patch the crate to use a newer upstream gettext version
+	__fetch_gettext_rs
+	__patch_gettext_rs
+}
+
+termux_step_make() {
+    termux_setup_rust
+
+    cd "${TERMUX_PKG_SRCDIR}"
+
+    RUSTFLAGS="-C link-arg=-Wl,-rpath=${TERMUX_PREFIX}/lib -C link-arg=-L${TERMUX_PREFIX}/lib -C link-arg=-liconv" \
+    cargo build \
+    --jobs "${TERMUX_PKG_MAKE_PROCESSES}" \
+    --target "${CARGO_TARGET_NAME}" \
+    --package rnote \
+    --release
+}
+
+termux_step_post_make_install() {
+    cd "${TERMUX_PKG_SRCDIR}"
+	install -Dm755 "target/$CARGO_TARGET_NAME/release/rnote" -t "$TERMUX_PREFIX/bin"
```

Comment:
Going by what is included in Arch Linux's `rnote` package we also want to include:
- `/bin/rnote-cli`
- `/share/fonts/rnote-fonts/*`
- `/share/glib-2.0/schemas/com.github.flxzt.rnote.gschema.xml`
- `/share/icons/hicolor/scalable/apps/com.github.flxzt.rnote.svg`
- `/share/icons/hicolor/scalable/mimetypes/application-rnote.svg`
- `/share/icons/hicolor/symbolic/apps/com.github.flxzt.rnote-symbolic.svg`
- `/share/locale/*/LC_MESSAGES/rnote.mo`
<sup>Although this may not apply since Android doesn't have proper locale support</sup>
- `/share/metainfo/com.github.flxzt.rnote.metainfo.xml`
- `/share/mime/packages/com.github.flxzt.rnote.xml`
- `/share/rnote/*`
*Mentally prepend `$TERMUX_PREFIX` to each of those locations*

I'll update these with the location they get compiled to later after I do a local build.

Reply:
There's no issue with including locales as that badcast issue was pinpointed to be caused by another crate(oboe-sys). Ironically enough, it's a crate that's responsible for audio

Reply:
> Going by what is included in Arch Linux's `rnote` package we also want to include:
> 
> * [x]  `/bin/rnote-cli`
> * [x]  `/share/fonts/rnote-fonts/*`
> * [x]  `/share/glib-2.0/schemas/com.github.flxzt.rnote.gschema.xml`
> * [x]  `/share/icons/hicolor/scalable/apps/com.github.flxzt.rnote.svg`
> * [x]  `/share/icons/hicolor/scalable/mimetypes/application-rnote.svg`
> * [x]  `/share/icons/hicolor/symbolic/apps/com.github.flxzt.rnote-symbolic.svg`
> * [ ]  `/share/locale/*/LC_MESSAGES/rnote.mo`
>   Although this may not apply since Android doesn't have proper locale support
> * [x]  `/share/metainfo/com.github.flxzt.rnote.metainfo.xml`
> * [x]  `/share/mime/packages/com.github.flxzt.rnote.xml`
> * [ ]  `/share/rnote/*`
>   This is where all of rnote's sound assets are stored
> 
> I'll update these with the location they get compiled to later after I do a local build.


Sound files should now be placed in the right directory
> * [x]  `/share/rnote/*`
``install -Dm755 "${TERMUX_PKG_SRCDIR}/crates/rnote-engine/data/sounds" -t "$TERMUX_PREFIX/share/rnote"``

I am skipping locales for now as I have no idea how to build them

Reply:
Well having the sound files doesn't do us any good if the sound support doesn't work.